enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,687 @@
+  uid: 99b64be3-4f36-4e0b-8704-75aa5182d89f
+  labels:
+    routeAdvertisements: default
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+  nodeSelector:
+    matchLabels:
+       kubernetes.io/hostname: ovn-worker
+```
+
+OVNKube-Controller will check that if this FRRConfiguration applies to its node, ovn-worker. For this example, a user-defined
+network named "blue", has been created with a network of 10.0.0.0/16, and a matching vrf exists in the Linux host. The slice
+of this supernet that has been allocated to node ovn-worker is 10.0.1./0/24. 
+
+##### Example 1: Advertising pod IPs from a user-defined network over BGP
+
+![](images/bgp_novpn_advertisements.png)
+
+In this example a user wants to expose network blue outside their OCP cluster so that pod IPs are reachable on the
+external network. The admin has creates the following RouteAdvertisements CR for the blue tenant:
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+  advertisements:
+    podNetwork: true
+  networkSelector:
+    matchLabels: 
+      k8s.ovn.org/metadata.name: blue
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/hostname: ovn-worker
+  frrConfigurationSelector:
+    matchLabels:
+      routeAdvertisements: default
+```
+
+OVNKube-Controller will now see it needs to generate corresponding FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: route-advertisements-blue
+  namespace: metallb-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      vrf: blue
+      prefixes:
+        - 10.0.1.0/24
+    - asn: 64512
+      neighbors:
+         - address: 172.18.0.5
+           asn: 64512
+           toAdvertise:
+              allowed:
+                 prefixes:
+                    - 10.0.1.0/24
+  raw:
+    rawConfig: |-
+       router bgp 64512
+        address-family ipv4 unicast
+          import vrf blue
+          exit-address-family
+       router bgp 64512 vrf blue    
+        address-family ipv4 unicast
+          import vrf default
+          exit-address-family      
+  nodeSelector:
+     matchLabels:
+        kubernetes.io/hostname: ovn-worker
+```
+
+In the above configuration generated by OVN-Kubernetes, the subnet 10.0.1.0/24 which belongs to VRF blue, is being
+imported into the default VRF, and advertised to the 172.18.0.5 neighbor. This is because the targetVRF was defaulted so
+the routes are leaked and advertised in the default VRF. Additionally, routes are being imported from the default VRF
+into the blue VRF.
+
+##### Example 2: VRF Lite - Advertising pod IPs from a user-defined network over BGP with VPN
```

Comment:
VRF-Lite is Cisco originated, but a widely understood name in the networking industry. It fits exactly what this use case is, where rather than using MPLS/EVPN + MP-BGP to encapsulate and carry VRFs we are using, separate IP links that map to a VRF and advertising separate dynamic routing sessions to the PE for routes to that VRF. I think VRF-Lite naming actually adds value, because customers are already familiar with this terminology.

Reply:
Ah, pardon my ignorance then!